www/include/reporting/dashboard/xmlInformations/common-Func.php

```diff
@@ -92,7 +92,7 @@ function fillBuffer($statesTab, $row, $color)
     $detailPopup = '{table class=bulleDashtab}';
     $detailPopup .= '{tr}{td class=bulleDashleft colspan=3}' . $Day .
         ': {span class ="isTimestamp isDate"}';
-    $detailPopup .= $date_start . '{/span} --  ' . $Duration . ': '.
+    $detailPopup .= date('Y-m-d', $date_start) . '{/span} --  ' . $Duration . ': ' .
```

Comment:
"timestamp" and "isDate" are classes keyword used by moment to overide and localize the timestamp in the customer format.
IMHO, you should try to find why moment isn't able to convert it, as it was introduced and working on the 18.10

Reply:
Are you sure about this @sc979 ?
I've been testing it in the latest 18.10.x, 19.04.x, 19.10.x and 20.04.x and it does not work :cry:

Reply:
Check the ticket MON-2491

Reply:
Sometimes validation can fail :+1: 
The detail popin was a subtask we might have skip its validation. With the actual code it cannot work. 

**Why ?**

Currently the formatDateMoment() function searchs in the DOM for elements defined with the `isTimestamp` class

formatDateMoment() --
```
----
    //creating a collection of occurrences
    jQuery(".isTimestamp").each(function (index, element) {
        var myElement = jQuery(element);
---
```
When loading the timeline **ie** when selecting an `id` in the SELECT (a hostgroup id for instance) the popins are not part of the DOM. So the `formatDateMoment()` will not format the popin.

We have checked with @kduret

We've tried to add a setTimeout after the loadXML, to wait for data to be loaded to call the `formatDateMoment()`

This is **not OK**
- The date gets translated after the time defined for the timeout which is weird
- The date gets translated only one time (when opening the popup). If we try to reopen the popin the date do not get translated.

To sum up at the Timeline level it is currently not possible to add eventListeners so we can't do this on the JS / FRONT side.

It needs to be done on backend side

Reply:
On the backend side, we will have to get the locale and Timezone on the PHP level (information available in the localStorage which can't be used on PHP).

We will have to get those values from the PHP session I guess.

WDYT @sc979 @kduret does it worth to do it knowing that none of the values are localized in the Timeline ... ?

Reply:
Except that, this feature was working fine. I've tested it before QA.
As you say, under certain conditions, you're able to make it work.
IMHO : you should investigate to find what introduced this regression

Reply:
A part of timeline library was overrided by Centreon, so finally I've updated this section to manage date conversion from frontend

Reply:
You rock